Frameworks/Foundation/NSURLConnection.mm

```diff
@@ -19,168 +20,333 @@
 #import "Foundation/NSMutableData.h"
 #import "Foundation/NSError.h"
 #import "Foundation/NSRunLoop.h"
-#import "NSURLConnectionState.h"
 #import "Foundation/NSURLConnection.h"
 #import "LoggingNative.h"
 #import "NSURLProtocolInternal.h"
 
+#import <algorithm>
+#import <vector>
+
 static const wchar_t* TAG = L"NSURLConnection";
 
-@implementation NSURLConnection
+// Private helper function that returns a NSURLProtocol object capable of handling a NSURLRequest
+// Returns nil if one cannot be created
+static NSURLProtocol* _protocolForRequest(NSURLRequest* request, id<NSURLProtocolClient> client) {
```

Comment:
Hmm, originally i only proposed this convention for within CoreText/Graphics, given that we had a Hhigh incidence of both file-private and framework-private functions. Did we begin to use this more broadly?